tidb-server/main.go

```diff
@@ -91,14 +98,36 @@ func main() {
 	printer.PrintTiDBInfo()
 	log.SetLevelByString(cfg.LogLevel)
 
-	store, err := tidb.NewStore(fmt.Sprintf("%s://%s", *store, *storePath))
+	fullPath := fmt.Sprintf("%s://%s", *store, *storePath)
+	u, err := url.Parse(fullPath)
+	if err != nil {
+		log.Fatal(errors.ErrorStack(err))
+	}
+	if cluster := u.Query().Get("cluster"); cluster != "" {
+		id, err1 := strconv.ParseUint(cluster, 10, 64)
+		if err1 != nil {
+			log.Fatal(errors.ErrorStack(err1))
+		}
+		binloginfo.ClusterID = id
+	}
```

Comment:
Can line 101-112 be extract into a function？I think  simpler and shorter code in the function of main is better. The same as line 122-130.